editor/dependency_editor.cpp

```diff
@@ -496,13 +496,21 @@ void DependencyRemoveDialog::ok_pressed() {
 			res->set_path("");
 		}
 
-		// If the file we are deleting is the main scene or default environment, clear its definition.
-		if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/run/main_scene")) {
+		// If the file we are deleting for e.g. the main scene or default environment, we must clear its definition in Project Settings.
+		if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/config/icon")) {
+			ProjectSettings::get_singleton()->set("application/config/icon", "");
+		} else if (files_to_delete[i] == ProjectSettings::get_singleton()->get("application/run/main_scene")) {
```

Comment:
Should be a plain `if` and not `else if`, as some files might be used by several properties (e.g. icon, cursor and boot splash could be the same).